massa-network/src/peer_info_database.rs

```diff
@@ -494,50 +451,27 @@ impl PeerInfoDatabase {
         if !ip.is_global() {
             return Err(NetworkError::InvalidIpError(*ip));
         }
-        let (available_bootstrap_conns, available_nonbootstrap_conns) =
-            self.get_available_out_connection_attempts();
+
```

Comment:
I think I see a problem here (that was probbably there even before).

Scenario;
* we ask for a peer list to connect to
* something causes peer cleanup, removing some of the peers from that list
* we notify new_out_connection_attempt to one of the peers we receeived at the 1st step => it will be refused

I think here, if the peer is not found, we should add it as a standard peer with 1 out connection attempt and call cleanup() at the end of the function (to remove any excess peers now that we have added one)

what do you think ?

Reply:
That makes sense.

Reply:
it appears that it's a tested behavior, so I don't know what to do about that, either revert the last commit 12c2f7397e91343ab0e95d9037fe1da40abe3203 and keep that reflection for a follow up or update the test according to the new behavior.

Reply:
ok let's keep it for a followup

Reply:
#2347